src/libraries/Microsoft.PowerFx.Core/Public/Config/SymbolTableOverRecordType.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
+
+namespace Microsoft.PowerFx
+{
+    // Wraps a record
+    // This also preserves the lazy field enumeration semantics of a recordType.
+    internal class SymbolTableOverRecordType : ReadOnlySymbolTable, IGlobalSymbolNameResolver
+    {
+        private readonly RecordType _type;
+
+        public SymbolTableOverRecordType(RecordType type)
+        {
+            _type = type;
+            _debugName = "per-eval";
+        }
+
+        IReadOnlyDictionary<string, NameLookupInfo> IGlobalSymbolNameResolver.GlobalSymbols
+        {
+            get
+            {
+                // $$$ Enumeration is for intellisense, should return in DisplayNames?
```

Comment:
>// $$$ Enumeration is for intellisense, should return in DisplayNames [](http://example.com/codeflow?start=16&length=69)

probably, or at least set the DisplayName in the NameLookupInfo constructed on line 55 and get intellisense to use that.